### PR TITLE
ADD: Added board support for Seeed Wio Terminal and Seeeduino XIAO

### PR DESCRIPTION
**Pull request by ansonhe97**
Hi, 

I've added board support for **Wio Terminal** and **Seeeduino XIAO**.  The current packages(ArduinoSAMD core) are installed using the git methods, maybe you can add it to the  `platform.json` list as officials.

* Wio Terminal's product page: https://www.seeedstudio.com/Wio-Terminal-p-4509.html

* Seeeduino XIAO's product page: https://www.seeedstudio.com/Seeeduino-XIAO-Arduino-Microcontroller-SAMD21-Cortex-M0+-p-4426.html

Thanks

### CODE REVIEW

File: builder/main.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def _jlink_cmd_script(env, source):
         ],
         UPLOADCMD="$UPLOADER $UPLOADERFLAGS $SOURCES"
     )
-    if board.get("build.core") == "adafruit" and board.get(
+    if ((board.get("build.core") == "adafruit") or (board.get("build.core") == "seeed")) and board.get(
```
Comment (valeros) [inline builder/main.py:217]: Please use the following syntax here and in the similar part of the code:
```python
if board.get("build.core") in ("adafruit", "seeed"):
    ...
```

File: builder/frameworks/arduino/arduino-samd.py [hunk 2]
```diff
@@ -78,7 +78,7 @@
         LIBS=["arm_cortexM0l_math"]
     )
 
-if BUILD_CORE in ("adafruit", "moteino"):
+if BUILD_CORE in ("seeed","adafruit", "moteino"):
```
Comment (valeros) [inline builder/frameworks/arduino/arduino-samd.py:81]: Please add a missing space after the comma.

File: boards/seeed_wio_terminal.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    "mcu": "samd51p19a",
+    "system": "samd",
+    "usb_product": "Seeed Wio Terminal",
+    "variant": "wio_terminal"
+  },
+  "debug": {
+    "jlink_device": "ATSAMD51P19",
+    "openocd_chipname": "at91samd51p19",
+    "openocd_target": "atsame5x",
+    "svd_path": "ATSAMD51P19A.svd"
```
Comment (valeros) [inline boards/seeed_wio_terminal.json:29]: There is no such SVD file in this repo.

File: boards/seeed_wio_terminal.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{
+  "build": {
+    "arduino": {
+        "ldscript": "flash_with_bootloader.ld"
+    },
+    "core": "seeed",
+    "cpu": "cortex-m4",
+    "extra_flags":"-D__SAMD51P19A__ -DWIO_TERMINAL -DSEEED_WIO_TERMINAL -DSEEED_GROVE_UI_WIRELESS -D__SAMD51__ -D__FPU_PRESENT -DARM_MATH_CM4 -mfloat-abi=hard -mfpu=fpv4-sp-d16",
```
Comment (valeros) [inline boards/seeed_wio_terminal.json:8]: Please use an array, for example, like here https://github.com/platformio/platform-atmelsam/blob/develop/boards/adafruit_feather_m4.json#L8-L17

**Comment by valeros** [comment]
Hi @ansonhe97 ! Many thanks for the PR! A few questions:
- Have you tried to upload firmware to these boards using PlatformIO?
- Are you sure that build flags are similar to the flags specified in the original platform.txt?
- What boards `Seeeduino Femto M0`, `Seeeduino Zero`, `Seeeduino LoRaWAN` and others? Are they supported?

I also added some comments to the code.

Thanks in advance!

File: boards/seeed_wio_terminal.json (hunk 4 above)
Comment (ansonhe97) [reply boards/seeed_wio_terminal.json]: These flags are used floating points, they are listed in the `platform.txt`.

File: boards/seeed_wio_terminal.json (hunk 3 above)
Comment (ansonhe97) [reply boards/seeed_wio_terminal.json:39]: Should i just remove the `svd path`?

File: boards/seeed_wio_terminal.json (hunk 4 above)
Comment (valeros) [reply boards/seeed_wio_terminal.json]: But they are also specified here https://github.com/platformio/platform-atmelsam/blob/develop/builder/frameworks/arduino/arduino-samd.py#L62-L72

File: boards/seeed_wio_terminal.json (hunk 3 above)
Comment (valeros) [reply boards/seeed_wio_terminal.json:39]: It'd be much better if you add this file

File: boards/seeed_wio_terminal.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      "-D__SAMD51P19A__",
+      "-DWIO_TERMINAL",
+      "-DSEEED_WIO_TERMINAL",
+      "-DSEEED_GROVE_UI_WIRELESS",
+      "-D__SAMD51__",
+      "-D__FPU_PRESENT",
+      "-DARM_MATH_CM4",
+      "-mfloat-abi=hard",
+      "-mfpu=fpv4-sp-d16"
+      ],
```
Comment (valeros) [inline boards/seeed_wio_terminal.json:18]: Indentation

File: boards/seeed_wio_terminal.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      ]
+    ],
+    "mcu": "samd51p19a",
+    "system": "samd",
+    "usb_product": "Seeed Wio Terminal",
+    "variant": "wio_terminal"
+  },
+  "debug": {
+    "jlink_device": "ATSAMD51P19",
+    "openocd_chipname": "at91samd51p19",
```
Comment (valeros) [inline boards/seeed_wio_terminal.json:37]: Also I'm not sure about this value, probably should be `at91samg5x`

File: boards/seeed_wio_terminal.json (hunk 4 above)
Comment (ansonhe97) [reply boards/seeed_wio_terminal.json]: Noted, will remove these flags here

File: boards/seeed_wio_terminal.json (hunk 3 above)
Comment (ansonhe97) [reply boards/seeed_wio_terminal.json:39]: Not sure where this svd_path actually points to, where do i add the file

Comment (valeros) [reply boards/seeed_wio_terminal.json:39]: https://github.com/platformio/platform-atmelsam/tree/develop/misc/svd

**Comment by ansonhe97** [comment]
Hi @valeros 

1. Yes I have tested uploading both WT and XIAO using PlatformIO.
2. Build flags are pretty much the same.
3. `Seeeduino Femto` is a special version of XIAO only sell in China(I just added), `LoraWAN` seems already supported.  

And `Seeeduino Zero` are for other SEEED boards similar to the Arduino Zero. This currently has some bugs with compiling errors  so will have a look again tomorrow :D

**Comment by ansonhe97** [comment]
Added support for `Seeeduino Zero` now :D

**Comment by ansonhe97** [comment]
All boards have been tested to compile and upload physically.

**Comment by valeros** [comment]
Awesome! Do you mind making the indentation consistent among all board manifests? I see some of them use 4 spaces, some use 2 spaces. Please use 2 spaces for indentation and please fix the new lines at the end of files for `boards/seeed_zero.json`, `boards/seeed_wio_lite_mg126.json`, `boards/seeed_femto.json`.

Thanks!

**Comment by ansonhe97** [comment]
Please check the above changes, let me know if I have missed something.

**Comment by valeros** [comment]
Thanks! I'll upload the Seeed's Arduino package to our CDN then merge the PR.

**Comment by ansonhe97** [comment]
Thanks!

**Comment by valeros** [comment]
`framework-arduino-samd-seeed` has been uploaded. Please delete the custom package from the examples:
```
platform_packages = framework-arduino-samd-seeed@https://github.com/Seeed-Studio/ArduinoCore-samd.git
```

**Comment by ansonhe97** [comment]
Done :D

**Comment by valeros** [comment]
Many thanks for your contribution!

**Pull request merged**